src/Enzo/enzo_EnzoMethodPmDeposit.cpp

```diff
@@ -97,221 +107,243 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
-    for (int i=0; i<mx*my*mz; i++) de_pa[i] = 0.0;
+      enzo_float dens = *((enzo_float *)(particle.constant_value (it,ic_mass)));
 
-    // check precisions match
+      // check precisions match
     
-    int ia = particle.attribute_index(it,"x");
-    int ba = particle.attribute_bytes(it,ia); // "bytes (actual)"
-    int be = sizeof(enzo_float);                // "bytes (expected)"
-
-    ASSERT4 ("EnzoMethodPmUpdate::compute()",
-	     "Particle type %s attribute %s defined as %s but expecting %s",
-	     particle.type_name(it).c_str(),
-	     particle.attribute_name(it,ia).c_str(),
-	     ((ba == 4) ? "single" : ((ba == 8) ? "double" : "quadruple")),
-	     ((be == 4) ? "single" : ((be == 8) ? "double" : "quadruple")),
-	     (ba == be));
-
-    // Accumulate particle density using CIC
-
-    enzo_float cosmo_a=1.0;
-    enzo_float cosmo_dadt=0.0;
-    EnzoPhysicsCosmology * cosmology = enzo::cosmology();
+      int ia = particle.attribute_index(it,"x");
+      int ba = particle.attribute_bytes(it,ia); // "bytes (actual)"
+      int be = sizeof(enzo_float);                // "bytes (expected)"
+
+      ASSERT4 ("EnzoMethodPmUpdate::compute()",
+               "Particle type %s attribute %s defined as %s but expecting %s",
+               particle.type_name(it).c_str(),
+               particle.attribute_name(it,ia).c_str(),
+               ((ba == 4) ? "single" : ((ba == 8) ? "double" : "quadruple")),
+               ((be == 4) ? "single" : ((be == 8) ? "double" : "quadruple")),
+               (ba == be));
+
+      // Accumulate particle density using CIC
+
+      enzo_float cosmo_a=1.0;
+      enzo_float cosmo_dadt=0.0;
+      EnzoPhysicsCosmology * cosmology = enzo::cosmology();
     
-    if (cosmology) {
+      if (cosmology) {
 
-      double time = block->time();
-      double dt   = block->dt();
-      cosmology->compute_expansion_factor (&cosmo_a,&cosmo_dadt,time+alpha_*dt);
+        double time = block->time();
+        double dt   = block->dt();
+        cosmology->compute_expansion_factor (&cosmo_a,&cosmo_dadt,time+alpha_*dt);
       
-    }
-    if (rank >= 1) hx *= cosmo_a;
-    if (rank >= 2) hy *= cosmo_a;
-    if (rank >= 3) hz *= cosmo_a;
+      }
+      if (rank >= 1) hx *= cosmo_a;
+      if (rank >= 2) hy *= cosmo_a;
+      if (rank >= 3) hz *= cosmo_a;
     
-    const double dt = alpha_ * block->dt() / cosmo_a;
-
-    // Scale mass by volume if particle value is mass instead of density
+      const double dt = alpha_ * block->dt() / cosmo_a;
 
-    int level = block->level();
+      // Scale mass by volume if particle value is mass instead of density
 
-    dens *= std::pow(2.0,rank*level);
+      int level = block->level();
 
-    // Accumulated single velocity array for Baryon deposit
-
-    for (int ib=0; ib<particle.num_batches(it); ib++) {
+      // Required for Cosmology ("mass" is mass)
+      // Not for Collapse ("mass" is density)
+      if (cosmology) {
+        dens *= std::pow(2.0,rank*level);
+      }
```

Comment:
@jobordner I'm ok with waiting to address this. But yes, making an issue for it would be good. I can imagine this being a potential non-obvious pitfall for someone down the line.